docs/earthfile/earthfile.md

```diff
@@ -684,6 +690,197 @@ Specifies the platform for any referenced `--load` and `--pull` images.
 
 For more information see the [multi-platform guide](../guides/multi-platform.md).
 
+## IF (**experimental**)
+
+{% hint style='danger' %}
+##### Important
+
+This feature is currently in **Experimental** stage
+
+* The feature may break, be changed drastically with no warning, or be removed altogether in future versions of Earthly.
+* Check the [GitHub tracking issue](https://github.com/earthly/earthly/issues/779) for any known problems.
+* Give us feedback on [Slack](https://earthly.dev/slack).
+{% endhint %}
+
+#### Synopsis
+
+* ```
+  IF [<condition-options>...] <condition>
+    <if-block>
+  END
+  ```
+* ```
+  IF [<condition-options>...] <condition>
+    <if-block>
+  ELSE
+    <else-block>
+  END
+  ```
+* ```
+  IF [<condition-options>...] <condition>
+    <if-block>
+  ELSE IF [<condition-options>...] <condition>
+    <else-if-block>
+  ...
+  ELSE
+    <else-block>
+  END
+  ```
+
+#### Description
+
+The `IF` clause can perform varying commands depending on the outcome of one or more conditions. The expression passed as part of `<condition>` is evaluated by running it in the build environment. If the exit code of the expression is zero, then the block of that condition is executed. Otherwise, the control continues to the next `ELSE IF` condition (if any), or if no condition returns a non-zero exit code, the control continues to executing the `<else-block>`, if one is provided.
+
+A very common pattern is to use the POSIX shell `[ ... ]` conditions. For example the following marks port `8080` as exposed if the file `./foo` exists.
+
+```Dockerfile
+IF [ -f ./foo ]
+  EXPOSE 8080
+END
+```
+
+{% hint style='info' %}
+##### Note
+Performing a condition requires that a `FROM` (or a from-like command, such as `LOCALLY`) has been issued before the condition itself.
+
+For example, the following is NOT a valid Earthfile.
+
+```Dockerfile
+# NOT A VALID EARTHFILE.
+ARG base=alpine
+IF [ "$base" = "alpine" ]
+    FROM alpine:3.13
+ELSE
+    FROM ubuntu:20.04
+END
+```
+
+The reason this is invalid is because the `IF` condition does not have a build environment initialized, to execute in.
```

Comment:
I wonder if we should explicitly call out `/usr/bin/[` to explain what's actually happening; not all users realise that `[` is actually a program.
```suggestion
The reason this is invalid is because the `IF` condition is actually running the `/usr/bin/[` executable to test if the condition is true or false, and therefore requires a valid build environment has been initialized.
```